### ISSUE TITLE
Could somebody add a "streaming audio" feature to Open ToonZ?

### ISSUE THREAD

**Issue by MaAlGon3** [opentoonz/opentoonz#133]
When I add a sound clip to the xsheet, I do so with the intention of being able to sync the sound with the animation I'm making. But as far as I've read in the forums, guides and tutorials, Open ToonZ does not have the ability to play the audio in your scene while you're scrolling through frames. It would be an incredibly useful feature and I'd really appreciate if the development team added it to the application.

**Comment by goldfish911** [comment]
What I and several others have found is that from the "X-sheet" tab, by left-click-dragging your mouse through the frame numbers on the far left on the main X-sheet, you can scrub through portions of the audio to work on syncing your animation.  However, I do not believe you can scrub the animation itself simultaneously. Hope this tip helps!

**Comment by 2dvision** [comment]
when streaming audio realtime, animator can hear what happened in animation. At the moment, i render video and look at it in VLC player.

**Comment by animking** [comment]
Please add this feature, it's really necessary to speed up the workflow.

**Comment by turtletooth** [comment]
@animking I will look into this.

**Comment by Rayek** [comment]
OT has three ways to scrub audio (from the Toonz Premum manual): 

1. Drag the xsheet frame cursor up or down to scrub all the audio columns whose Camera Stand toggle is active.
2. Drag the frame cursor in the viewer framebar to scrub all the audio columns whose Camera Stand toggle is active. (this does NOT work in the comboviewer - only the viewer)
3. Windows only: click and drag on the dashed vertical strip available on the right of the audio column cells: the selected section will be automatically played back.

In my opinion we should amend this with two new options:

1. a checkbox in the play control bar which activates sound rendering while using the regular playback commands such as looping and the play buttons;
2. when the previous and next frame commands are mapped to shortcut keys, and pressing and holding down those keys one would expect to be able to playback the audio. Currently only fragments are played back, making this useless for real work.

In short: allow audio playback when using the regular play controls, and when using keys to control audio playback.

**Comment by animking** [comment]
Yes we just need sound playback in default viewer same as the preview window does :)

**Comment by turtletooth** [comment]
Fixed in #788

**Comment by prinxe210** [comment]
Im new to this program and im starting to get the hang of it the only issue i have right now is that im having trouble importing my music into the project. I have a screen shot of the message I keep getting because its hard to explain in my own words but this is what i've been getting
![screenshot 21](https://user-images.githubusercontent.com/29809751/27763864-b07a3b88-5e51-11e7-849c-1be768403dbc.png)

**Comment by RodneyBaker** [comment]
Duplicate post, answered in another post and as this relates to a standard import/loading feature.
Questions related to normal operation of OpenToonz can be posted via the google group forum:

https://groups.google.com/forum/#!forum/opentoonz_en
